mflowgen/test/filters/mentor-calibre-lvs.awk

```diff
@@ -0,0 +1,101 @@
+# Filter for lvs buildkite log
+
+# Don't start filter until step begins
+BEGIN { PRINTALL=1 }
+/Checking preconditions for .*mentor-calibre-lvs/ { PRINTALL=0 }
+# PRINTALL { print "PA " $0; next }
+PRINTALL { print "PA " $0; next }
+
+# Show end matter
+/LVS completed/    { PRINTALL=1 }
+
+# Show max 20 warnings
+{
+    IGNORECASE = 1
+    if ($0 ~ /warn/) {
+        nwarns++; if (nwarns < 20) {
+            # print "FOO " $0
+            print "WA " $0
+            next
+        }
+    }
+    IGNORECASE = 0
+}
+
+# Show ALL warning/error/pass/fail lines
+{
+    IGNORECASE = 1
+    if      ($0 ~ /error/) { print "WE " $0; next }
+    else if ($0 ~ /pass/)  { 
+        if ($0 ~ /YPASS/) { next } ; # Ignore false positives :(
+        print "WE " $0; next 
+    }
+    else if ($0 ~ /fail/)  { print "WE " $0; next }
+    IGNORECASE = 0
+}
+
+# Show timedesign summary
+BEGIN { TDSUMMARY=0 }
+/Profile/ { TDSUMMARY=0 }
+TDSUMMARY==1 { print; next }
+/timeDesign Summary/ {
+    TDSUMMARY=1
+    print "------------------------------------------------------------"
+    print $0
+    next
+}
+# ------------------------------------------------------------
+#           timeDesign Summary                             
+# ------------------------------------------------------------
+# 
+# Hold  views included:
+#  analysis_default
+# 
+# +--------------------+---------+---------+---------+---------+
+# |     Hold mode      |   all   | reg2reg |reg2cgate| default |
+# +--------------------+---------+---------+---------+---------+
+# |           WNS (ns):| -0.612  | -0.612  | -0.024  |  0.611  |
+# |           TNS (ns):| -2.282  | -2.040  | -0.242  |  0.000  |
+# |    Violating Paths:|   93    |   71    |   22    |    0    |
+# |          All Paths:|  81113  |  76601  |  3846   |   756   |
+# +--------------------+---------+---------+---------+---------+
+# |analysis_default    | -0.612  | -0.612  | -0.024  |  0.611  |
+# |                    | -2.282  | -2.040  | -0.242  |  0.000  |
+# |                    |   93    |   71    |   22    |    0    |
+# |                    |  81113  |  76601  |  3846   |   756   |
+# +--------------------+---------+---------+---------+---------+
+# 
+# Density: 1.179%
+#        (376.862% with Fillers)
+# ------------------------------------------------------------
+# ** Profile ** Report data :  cpu=0:02:44, mem=12376.0M
+
+# Rolling buffer of last ten not-yet-printed lines
```

Comment:
Why is all this timing stuff in an LVS filter? This is not Calibre output.

Reply:
Hm good point. But for now, it hurts nothing to have this in there and so maybe shouldn't block the pull; meanwhile I'll try and figure out why it's there and if/when I should delete it...

Reply:
What is it trying to accomplish right now?

Reply:
This filter runs whenever the user requests '--steps lvs'. Thus it filters all the steps along the way to lvs; synthesis, place, route etc. If / when one or more of those steps produces a timing summary, this will pass that summary through to the buildkite log. This may or may not be the best way to do that. But (I think) it'll do til something better comes along :)

Reply:
Oh.. well in that case it should probably be named something else. Seems like it's just a general filter for any full chip run.

Reply:
> Seems like it's just a general filter for any full chip run.

This filter *only* runs if the user specifies `--steps lvs` on the command line. At which point it filters everything that happens as a result of `make lvs` including e.g. synthesis, place and route, i.e. it's basically doing `make lvs | lvs-filter.awk`. If the user specifies e.g. `--steps syn,place,route,lvs`, then the test script will run `make synthesis; make place; make route; make lvs | lvs-filter.awk`.

Eventually there are supposed to be more filters e.g. syn.awk, place.awk, route.awk etc. (all optional of course), this is just a starting point / example. To be generally useful, each filter in the chain will need to be a superset of all the filters lower down in the chain. There are probably better ways of doing this, and hopefully this will evolve toward the better ways.

But for now, this does what I want it to do...